src/users/components/ApproveButton.vue

```diff
@@ -0,0 +1,243 @@
+    :color="hasRole ? 'secondary' : 'white'"
+    :text-color="hasRole ? 'white' : 'grey-8'"
+    :title="tooltip"
+    icon="fas fa-leaf"
+    @click="showing = true"
+  >
+    <QDialog
+      v-model="showing"
+    >
+      <QCard>
+        <QCardSection class="row no-wrap justify-between">
+          <div class="text-h6">
+            {{ headline }}
+          </div>
+        </QCardSection>
+        <QCardSection
+          style="max-width: 700px"
+        >
+          <p>{{ info }}</p>
+          <p>{{ role.description }}</p>
+        </QCardSection>
+        <QCardSection>
+          <ProfilePicture
+            v-for="u in trustedBy"
+            :key="u.id"
+            :user="u"
+            :size="35"
+            class="q-mr-sm"
+          />
+        </QCardSection>
+        <QCardActions align="right">
+          <QBtn
+            v-close-popup
+            flat
+            :label="$t('BUTTON.CLOSE')"
+          />
+          <QBtn
+            v-if="!isTrusted && !isCurrentUser"
+            color="secondary"
+            :loading="trustUserStatus.pending"
+            @click="showCreateTrustDialog"
+          >
+            <span class="q-mr-xs">+</span>
+            <QIcon name="fas fa-leaf" />
+          </QBtn>
+          <QBtn
+            v-if="isTrusted && !isCurrentUser"
+            color="negative"
+            :loading="revokeTrustStatus.pending"
+            @click="showRevokeTrustDialog"
+          >
+            <span class="q-mr-xs">-</span>
+            <QIcon name="fas fa-leaf" />
+          </QBtn>
+        </QCardActions>
+      </QCard>
+    </QDialog>
+  </QBtn>
+</template>
+
+<script>
+import {
+  QBtn,
+  QDialog,
+  QCard,
+  QCardSection,
+  QCardActions,
+  QIcon,
+  Dialog,
+} from 'quasar'
+import { toRefs, computed } from 'vue'
+
+import { useAuthHelpers } from '@/authuser/helpers'
+import { useAuthService } from '@/authuser/services'
+import { useRevokeTrustMutation, useTrustUserMutation } from '@/group/mutations'
+import { useCurrentGroupService } from '@/group/services'
+import { useUserService } from '@/users/services'
+
+import ProfilePicture from '@/users/components/ProfilePicture'
+
+export default {
+  components: {
+    ProfilePicture,
+    QBtn,
+    QDialog,
+    QCard,
+    QCardSection,
+    QCardActions,
+    QIcon,
+  },
+  props: {
+    user: {
+      default: null,
+      type: Object,
+    },
+    membership: {
+      default: () => ({}),
+      type: Object,
+    },
+  },
+  setup (props) {
+    const {
+      user,
+      membership,
+    } = toRefs(props)
+
+    const {
+      group,
+      groupId,
+    } = useCurrentGroupService()
+
+    const {
+      getUserById,
+    } = useUserService()
+
+    const {
+      getIsCurrentUser,
+    } = useAuthHelpers()
+
+    const {
+      userId: authUserId,
+    } = useAuthService()
+
+    const isCurrentUser = computed(() => getIsCurrentUser(user.value))
+    const hasRole = computed(() => membership.value.roles.includes('approved'))
```

Comment:
This component seems _mostly_ generic, but then specific to `approved`. I wonder how far off it is to have one trust button that can be used for editor or approved role? Seems close?

Reply:
"trust for role" is still somewhat WIP right now, I guess when it has been tested more in production we can replace the editor trust button with it.

My idea was also to have a different component, with less popups, then scales better with more roles. The current implementation would just add more round buttons with unclear icons next to each other ;) Rather, I would have a section in the profile page that has some space for text and explanations.